### ISSUE TITLE
input_format_import_nested_json doesn't work for non array types

### ISSUE THREAD

**Issue by minaevmike** [yandex/ClickHouse#3289]
i am trying to use `JSONEachRow` format with `input_format_import_nested_json` option to import non plain json from kafka. So i am trying to do simple thing for testing only
```
clickhouse-client --host 0.0.0.0 -m --input_format_import_nested_json=1
CREATE TABLE json_noisy (d1 UInt8, d2 String, b Nested(v String)) ENGINE = Memory ;
insert into json_noisy format JSONEachRow {"d2":"ok", "d1":1, "b": {"v": "1"}};
Exception on client:
Code: 27. DB::Exception: Cannot parse input: expected [ before: "1"}};: (while read the value of key b.v): (at row 1)
```
Seems parser expects that value of `v.b` would be array(seems because `b` is a nested table). So i don't know is it a bug or feature request, but it would very usefull if it would be possible to insert non array types into nested fields. Or may be there is some workaround for this problem?

**Comment by 4ertus2** [comment]
JSONEachRow is a simple json-like format that optimized for perfomance. It doesn't support complex nested structs. It's possible to insert Nested type via array syntax:

```
CREATE TABLE test.json_each_row_nested(
n Nested (s String, i Int32)
) ENGINE = Memory
```
{ "n.i":[45, 67, 8], "n.s":["x", "y", "z"] }

But It's not possible to insert named fields.

**Comment by filimonov** [comment]
PR #3144 introduce support for nested structs when subkeys are arrays, but it's not merged yet and i have no idea how should in deal with json structure you show.

May be @veloman-yunkan have some ideas regarding that?

**Comment by champtar** [comment]
There is a fix for #3144 that is not merged yet: #3223

**Comment by georgiopandarez** [comment]
I ran into this today.

JSONEachRow format with input_format_import_nested_json won't JSON looking like {"a": {"b": 123 } which makes it kinda useless with a Nested schema.
